src/core/crypto/impl/cryptopp/radix.cc

```diff
@@ -109,16 +140,8 @@ std::vector<std::uint8_t> Base64::Decode(
     const std::uint64_t len)
 {
   // Prepare decoder
-  int lookup[256];
-  CryptoPP::Base64Decoder::InitializeDecodingLookupArray(
-      lookup,
-      reinterpret_cast<const CryptoPP::byte*>(m_Base64Alphabet.c_str()),
-      GetAlphabet().size(),
-      false);
-
-  CryptoPP::AlgorithmParameters const params(CryptoPP::MakeParameters(
-      CryptoPP::Name::DecodingLookupArray(),
-      reinterpret_cast<const int*>(lookup)));
+  CryptoPP::AlgorithmParameters static const params(CryptoPP::MakeParameters(
+      CryptoPP::Name::DecodingLookupArray(), static_cast<const int*>(s_Base64Table), false));
```

Comment:
>reinterpret_cast<const CryptoPP::byte*>(m_Base64Alphabet.c_str()),
>GetAlphabet().size(),

@noloader Live tests pass and unit-tests pass but what I don't understand is how they are able to pass without using our custom alphabet. I'm concerned about silent failures but our [testcase](https://github.com/monero-project/kovri/blob/master/tests/unit_tests/core/crypto/radix.cc#L133) shows successful decoding. How is this possible?!

Reply:
> what I don't understand is how they are able to pass without using our custom alphabet.

It did use your custom alphabet:

```
p$ cat test.cxx
#include "cryptlib.h"
#include "secblock.h"
#include "osrng.h"
#include "base32.h"
#include "hex.h"

#include <iostream>
#include <string>

using namespace CryptoPP;

int main(int argc, char* argv[])
{
    AutoSeededRandomPool prng;

    // char alphabet[] = "abcdefghijklmnopqrstuvwxyz234567";  // and 32, true
    char alphabet[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-~";

    int lookup[256];
    Base32Decoder::InitializeDecodingLookupArray(
        lookup,
        (const byte*)alphabet, 64,
        false);

    unsigned int i=0;
    while(i<COUNTOF(lookup))
    {
        std::cout << lookup[i] << ", ";

        if (++i%16 == 0)
            std::cout << "\n";
    }
    std::cout << std::endl;

    // Test for Monero usage
    AlgorithmParameters static const params(MakeParameters(
        Name::DecodingLookupArray(), static_cast<const int*>(lookup), false));
    return 0;
```

Reply:
Sorry! I brain farted when comparing the RFC table. Thanks for clarifying.